### ISSUE TITLE
 Error: "erigontech/mdbx-go@v0.27.24/mdbx/env.go:110:22: could not determine kind of name for C.__va_list_tag"

### ISSUE THREAD

**Issue by SipengXie** [ledgerwatch/erigon#9839]
Description:
I encountered an error while building the program with the following message: `erigontech/mdbx-go@v0.27.24/mdbx/env.go:110:22: could not determine kind of name for C.__va_list_tag`. I'm using Go version 1.20.12 and have set the environment variable CC="clang".

Could you please guide me on how to resolve this issue?

Please let me know if there's any additional information needed. Thank you!

Best regards.

**Comment by AskAlexSharov** [comment]
`go env`

**Comment by SipengXie** [comment]
```shell
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/wulab/.cache/go-build"
GOENV="/home/wulab/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/wulab/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/wulab/go"
GOPRIVATE=""
GOPROXY="direct"
GOROOT="/usr/lib/golang"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.20.12"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="clang"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/wulab/interact/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -Wl,--no-gc-sections -fmessage-length=0 -gno-record-gcc-switches"
```

**Comment by AskAlexSharov** [comment]
git status
go build -v ./cmd/erigon

**Comment by SipengXie** [comment]
The result of `git status`:
``` shell
Not currently on any branch.
nothing to commit, working tree clean
```

The result of `git log`:
``` shell
commit a013ec25ccc2f42a6948638f0f6bd594bfd55b48 (HEAD, tag: v2.59.2, origin/release/2.59)
Author: Andrew Ashikhmin <34320705+yperbasis@users.noreply.github.com>
Date:   Thu Mar 28 12:25:06 2024 +0100

    Revert "updateForkChoice: support --sync.loop.block.limit (#9315)" (#9833)

    Revert PR #9315 since it apparently causes MDBX bloat and performance
    regression. The regression was introduced in
    [v2.58.0](https://github.com/ledgerwatch/erigon/releases/tag/v2.58.0).
```

The result of `go builod -v ./cmd/erigon':
```shell
...
# vendor/github.com/golang-fips/openssl-fips/openssl
openssl_port_aead_gcm.c:23:15: warning: assigning to 'uint8_t *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
openssl_port_aead_gcm.c:27:9: warning: assigning to 'uint8_t *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
openssl_port_aead_gcm.c:112:9: warning: assigning to 'uint8_t *' (aka 'unsigned char *') from 'char [1]' converts between pointers to integer types with different sign [-Wpointer-sign]
...
# github.com/erigontech/mdbx-go/mdbx
../../go/pkg/mod/github.com/erigontech/mdbx-go@v0.27.24/mdbx/env.go:110:22: could not determine kind of name for C.__va_list_tag
...

```

**Comment by AskAlexSharov** [comment]
I don’t know. 
try: go clean -cache -modcache
if will not help: remove and install golang again (when clan already installed). I guess it miss-configured a bit.

**Comment by SipengXie** [comment]
Thank u, I'll give it a try and then come back and give you feedback.

**Comment by SipengXie** [comment]
still gets ` github.com/erigontech/mdbx-go/mdbx
../../go/pkg/mod/github.com/erigontech/mdbx-go@v0.27.24/mdbx/env.go:110:22: could not determine kind of name for C.__va_list_tag` after re-installing updating  golang to 1.22.1

**Comment by AskAlexSharov** [comment]
i don't know. something miss-configured on server. cc -v ? OS? do you change to clang for fun? 

I usually install gcc on linux next way:
```
sudo apt install -y gcc-13 g++-13 golang-1.21-go
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-13 100 --slave /usr/bin/g++ g++ /usr/bin/g++-13 --slave /usr/bin/gcov gcov /usr/bin/gcov-13
sudo ln -s /usr/lib/go-1.21/bin/go /usr/bin/go
```
maybe you also need update-alternatives

**Comment by SipengXie** [comment]
Thank you! Let me try again.

**Comment by SipengXie** [comment]
Solved by re-installing my OS...
I will close this issue.
